lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py

```diff
@@ -380,6 +380,18 @@ def wait_for_event(self, filter=None, timeout=None):
             )
         return None
 
+    def wait_for_events(self, events, timeout=None):
+        """Wait for a list of events in `events` in any order.
+        Return the events not hit before the timeout expired"""
+        events = events[:]  # Make a copy to avoid modifying the input
+        end_time = time.time() + timeout if timeout else None
+        while events and (not end_time or end_time > time.time()):
+            event_dict = self.wait_for_event(filter=events, timeout=timeout)
+            if not event_dict:
+                continue
+            events.remove(event_dict["event"])
```

Comment:
Would this be equivalent?
```suggestion
        while events:
            event_dict = self.wait_for_event(filter=events, timeout=timeout)
            if event_dict is None:
                # Timeout
                break	
            events.remove(event_dict["event"])
```

Reply:
@JDevlieghere I think its not quite the same because it is allowing a timeout per-event instead of a "global" timeout for all the events. So we could end up with around `len(events)*timeout` of total time if each event is triggered just before the timeout expires.

In @kusmour version it would be bounder closer to `2 * timeout` if the first event is triggered just before the timeout expires.

I think having the timeout be per-event and not global is reasonable and simplifies the code a bit.

Reply:
Yeah, I should have phrased it differently. What I meant was that the current code already doesn't quite do what you'd expect (as you pointed out: it waits longer than the timeout), in which case I think waiting up to `timeout` for each event is more predictable. The other alternative is to wait up to n times for `timeout/n` but I think that also needless complicates the code.

Reply:
It should be fine. But in this case we reset the timeout on every call waiting for any single event in the filter. So it's possible that it takes longer than the `timeout` to get all events.